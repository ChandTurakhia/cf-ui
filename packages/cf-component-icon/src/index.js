import IconUnstyled from './Icon';
import IconTheme from './IconTheme';

import { applyTheme } from 'cf-style-container';

const Icon = applyTheme(IconUnstyled, IconTheme);

export { Icon, IconUnstyled, IconTheme };
