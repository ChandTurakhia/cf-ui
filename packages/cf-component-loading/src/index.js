import LoadingUnstyled from './Loading';
import LoadingTheme from './LoadingTheme';

import { applyTheme } from 'cf-style-container';

const Loading = applyTheme(LoadingUnstyled, LoadingTheme);

export { Loading, LoadingUnstyled, LoadingTheme };
