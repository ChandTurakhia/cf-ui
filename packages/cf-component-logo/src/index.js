import Logo from './Logo';
import logoSvg from '../assets/logo.svg';
import logoSvgMobile from '../assets/logo_mobile.svg';

export { Logo, logoSvg, logoSvgMobile };
