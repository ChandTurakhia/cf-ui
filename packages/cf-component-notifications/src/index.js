import NotificationUnstyled from './Notification';
import NotificationListUnstyled from './NotificationList';
import NotificationGlobalContainerUnstyled from './NotificationGlobalContainer';

import NotificationTheme from './NotificationTheme';
import NotificationListTheme from './NotificationListTheme';
import NotificationGlobalContainerTheme
  from './NotificationGlobalContainerTheme';

import { applyTheme } from 'cf-style-container';

const Notification = applyTheme(NotificationUnstyled, NotificationTheme);

const NotificationList = applyTheme(
  NotificationListUnstyled,
  NotificationListTheme
);

const NotificationGlobalContainer = applyTheme(
  NotificationGlobalContainerUnstyled,
  NotificationGlobalContainerTheme
);

export {
  Notification,
  NotificationList,
  NotificationGlobalContainer,
  NotificationUnstyled,
  NotificationListUnstyled,
  NotificationGlobalContainerUnstyled,
  NotificationTheme,
  NotificationListTheme,
  NotificationGlobalContainerTheme
};
