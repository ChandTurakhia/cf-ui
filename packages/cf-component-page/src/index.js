import PageUnstyled from './Page';
import PageHeaderUnstyled from './PageHeader';
import PageContentUnstyled from './PageContent';
import PageTheme from './PageTheme';
import Header from './Header';
import Logo from './Logo';
import NavList from './NavList';
import NavItem from './NavItem';
import logoSvg from './logo.svg';
import logoSvgMobile from './logo_mobile.svg';

import { applyTheme } from 'cf-style-container';

const Page = applyTheme(PageUnstyled, PageTheme);
const PageHeader = applyTheme(PageHeaderUnstyled, {});
const PageContent = applyTheme(PageContentUnstyled, {});

export {
  Header,
  Logo,
  logoSvg,
  logoSvgMobile,
  NavItem,
  NavList,
  Page,
  PageTheme,
  PageUnstyled,
  PageHeader,
  PageHeaderUnstyled,
  PageContent,
  PageContentUnstyled
};
