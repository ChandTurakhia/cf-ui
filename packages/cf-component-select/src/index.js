import { applyTheme } from 'cf-style-container';

import SelectUnstyled from './Select';
import SelectTheme from './SelectTheme';

const Select = applyTheme(SelectUnstyled, SelectTheme);

export { Select, SelectUnstyled, SelectTheme };
