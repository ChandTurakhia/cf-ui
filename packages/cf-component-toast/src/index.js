import { applyTheme } from 'cf-style-container';

import ToastUnstyled from './Toast';
import ToastTheme from './ToastTheme';

const Toast = applyTheme(ToastUnstyled, ToastTheme);

export { Toast, ToastUnstyled, ToastTheme };
