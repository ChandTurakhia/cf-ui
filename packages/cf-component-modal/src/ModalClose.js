import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'cf-component-icon';

class ModalClose extends React.Component {
  render() {
    return (
      <span className="cf-modal__close" onClick={this.props.onClick}>
        <Icon type="remove" label={this.props.label} />
      </span>
    );
  }
}

ModalClose.propTypes = {
  onClick: PropTypes.func.isRequired,
  label: PropTypes.oneOfType([PropTypes.string, PropTypes.bool])
};

ModalClose.defaultProps = {
  label: false
};

export default ModalClose;
