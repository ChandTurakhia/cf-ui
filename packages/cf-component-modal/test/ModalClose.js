import React from 'react';
import { felaSnapshot } from 'cf-style-provider';
import { ModalClose } from '../../cf-component-modal/src/index';

test('should render', () => {
  const snapshot = felaSnapshot(<ModalClose onClick={() => {}} />);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});

test('should render with a label', () => {
  const snapshot = felaSnapshot(
    <ModalClose label="label" onClick={() => {}} />
  );
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});
