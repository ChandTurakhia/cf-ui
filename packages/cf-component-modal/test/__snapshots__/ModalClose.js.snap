// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<span
  className="cf-modal__close"
  onClick={[Function]}
>
  <svg
    aria-label=""
    className="a b"
    viewBox="0 0 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <polygon
      points="14.3 3.625 12.375 1.7 8 6.075 3.625 1.7 1.7 3.625 6.075 8 1.7 12.375 3.625 14.3 8 9.925 12.375 14.3 14.3 12.375 9.925 8 14.3 3.625"
    />
  </svg>
</span>
`;

exports[`should render 2`] = `
"
.a {
  height: 15px
}

.b {
  fill: #000
}
"
`;

exports[`should render with a label 1`] = `
<span
  className="cf-modal__close"
  onClick={[Function]}
>
  <svg
    aria-label="label"
    className="a b"
    viewBox="0 0 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <polygon
      points="14.3 3.625 12.375 1.7 8 6.075 3.625 1.7 1.7 3.625 6.075 8 1.7 12.375 3.625 14.3 8 9.925 12.375 14.3 14.3 12.375 9.925 8 14.3 3.625"
    />
  </svg>
</span>
`;

exports[`should render with a label 2`] = `
"
.a {
  height: 15px
}

.b {
  fill: #000
}
"
`;
